webapp/sources/rudder/rudder-web/src/main/elm/onboarding/sources/DataTypes.elm

```diff
@@ -4,17 +4,22 @@ import Http exposing (Error)
 import List exposing (..)
 
 type Section
-  = Welcome SectionState
+  = Welcome
   | Account SectionState AccountSettings
-  | Metrics SectionState String
+  | Metrics SectionState MetricsState
   | GettingStarted SectionState
 
 type SectionState
   = Default
   | Visited
-  | Complete
+  | Completed
   | Warning
 
+type MetricsState
```

Comment:
There is four state, especially when running Rudder for the first time the value is "not_defined" 

either you need a 4 state MetricState (NotDefined) or have a (Maybe MetricsState)

I would go for a 4th state "NotDefined" and use to determine if the section was completed or not